x/migration/types/message_claim_morse_application_test.go

```diff
@@ -1,29 +1,106 @@
 	"testing"
 
 	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
-	"github.com/pokt-network/poktroll/testutil/sample"
 	"github.com/stretchr/testify/require"
+
+	"github.com/pokt-network/poktroll/testutil/sample"
+	apptypes "github.com/pokt-network/poktroll/x/application/types"
+	migrationtypes "github.com/pokt-network/poktroll/x/migration/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
 )
 
+const testServiceId = "svc1"
+
 func TestMsgClaimMorseApplication_ValidateBasic(t *testing.T) {
 	tests := []struct {
 		name string
-		msg  MsgClaimMorseApplication
+		msg  migrationtypes.MsgClaimMorseApplication
 		err  error
 	}{
 		{
-			name: "invalid address",
-			msg: MsgClaimMorseApplication{
+			name: "invalid ShannonDestAddress",
+			msg: migrationtypes.MsgClaimMorseApplication{
 				ShannonDestAddress: "invalid_address",
+				MorseSrcAddress:    sample.MorseAddressHex(),
+				MorseSignature:     mockMorseSignature,
+				Stake:              &apptypes.DefaultMinStake,
+				ServiceConfig: &sharedtypes.ApplicationServiceConfig{
+					ServiceId: testServiceId,
+				},
 			},
 			err: sdkerrors.ErrInvalidAddress,
 		}, {
-			name: "valid address",
-			msg: MsgClaimMorseApplication{
+			name: "invalid MorseSrcAddress",
+			msg: migrationtypes.MsgClaimMorseApplication{
+				ShannonDestAddress: sample.AccAddress(),
+				MorseSrcAddress:    "invalid_address",
+				MorseSignature:     mockMorseSignature,
+				Stake:              &apptypes.DefaultMinStake,
+				ServiceConfig: &sharedtypes.ApplicationServiceConfig{
+					ServiceId: testServiceId,
+				},
+			},
+			err: migrationtypes.ErrMorseApplicationClaim,
+		}, {
+			name: "invalid service ID (empty)",
+			msg: migrationtypes.MsgClaimMorseApplication{
+				ShannonDestAddress: sample.AccAddress(),
+				MorseSrcAddress:    sample.MorseAddressHex(),
+				MorseSignature:     mockMorseSignature,
+				Stake:              &apptypes.DefaultMinStake,
+				ServiceConfig: &sharedtypes.ApplicationServiceConfig{
+					ServiceId: "",
+				},
+			},
+			err: migrationtypes.ErrMorseApplicationClaim,
+		}, {
+			name: "invalid service ID (too long)",
+			msg: migrationtypes.MsgClaimMorseApplication{
+				ShannonDestAddress: sample.AccAddress(),
+				MorseSrcAddress:    sample.MorseAddressHex(),
+				MorseSignature:     mockMorseSignature,
+				Stake:              &apptypes.DefaultMinStake,
+				ServiceConfig: &sharedtypes.ApplicationServiceConfig{
+					ServiceId: "xxxxxxxxxxxxxxxxxxxx",
+				},
+			},
+			err: migrationtypes.ErrMorseApplicationClaim,
+		}, {
+			name: "invalid empty MorseSignature",
+			msg: migrationtypes.MsgClaimMorseApplication{
+				ShannonDestAddress: sample.AccAddress(),
+				MorseSrcAddress:    sample.MorseAddressHex(),
+				MorseSignature:     nil,
+				Stake:              nil,
+				ServiceConfig: &sharedtypes.ApplicationServiceConfig{
+					ServiceId: testServiceId,
+				},
+			},
+			err: migrationtypes.ErrMorseApplicationClaim,
+		}, {
+			name: "valid nil stake",
```

Comment:
Should be invalid

Reply:
The stake field will be remove since we don't need to support arbitrary stake amounts (as per our offline discussion).